### ISSUE TITLE
`getDhtOpType`, `getDhtOpAction`, `getDhtOpEntry`, `getDhtOpSignature` functions are incorrect in v0.18.0-rc 

### ISSUE THREAD

**Issue by guillemcordoba** [holochain/holochain-client-js#306]
In holochain v0.4, the `DhtOp` type was changed to: 

```js
export type DhtOp =
  | {
      ChainOp: ChainOp;
    }
  | { WarrantOp: WarrantOp };
```

Which actually makes all the DhtOp helpers incorrect, these is how they look like: 

```js
export function getDhtOpEntry(op: DhtOp): Entry | undefined {
  return Object.values(op)[0][2];
}
```

So they don't throw a typescript error, but actually they won't work. They should probably accept a `ChainOp` argument instead of a `DhtOp` one.

**Comment by matthme** [comment]
It might still be useful not to have to filter by WarrantOp and ChainOp before passing it in? What about a type `DhtOpSubType` that can be either of the ChainOp types or `Warrant`?

**Comment by guillemcordoba** [comment]
Yeah some of them yeah, some of them no. WarrantOp only carries the ActionHash for the invalid record, which makes things like `getDhtOpType()` or `getDhtOpAction()` not work at all with it.

**Comment by matthme** [comment]
I was thinking that `getDhtOpType` would now return the following type:
```
export enum ChainOpSubType {
  StoreRecord = "StoreRecord",
  StoreEntry = "StoreEntry",
  RegisterAgentActivity = "RegisterAgentActivity",
  RegisterUpdatedContent = "RegisterUpdatedContent",
  RegisterUpdatedRecord = "RegisterUpdatedRecord",
  RegisterDeletedBy = "RegisterDeletedBy",
  RegisterDeletedEntryAction = "RegisterDeletedEntryAction",
  RegisterAddLink = "RegisterAddLink",
  RegisterRemoveLink = "RegisterRemoveLink",
  Warrant = "Warrant",
}
```
And `getDhtOpAction` could return undefined in the warrant case.

But maybe it's cleaner to just call it `getChainOpType` and `getChainOpAction` and take a ChainOp as you suggest and then the caller needs to filter explicityl beforehand.

**Comment by matthme** [comment]
Here's a PR: https://github.com/holochain/holochain-client-js/pull/308
What do you think @guillemcordoba? (Can't add you as reviewer in the PR itself apparently)

**Comment by matthme** [comment]
Fixed with #308 and published as 0.18.0-rc.2. But unfortunately I didn't see the other issue about the wrong `ChainIntegrityWarrant` type so this will be part of a next release...
